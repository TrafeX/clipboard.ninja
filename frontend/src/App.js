import React, { useEffect } from 'react';
import { orange600, orange800, blueGrey400 } from 'material-ui/styles/colors';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'; // v1.x
import { MuiThemeProvider as V0MuiThemeProvider } from 'material-ui';
import { Helmet } from "react-helmet";
import ReactGA from 'react-ga';
import { Route, Switch, useHistory } from 'react-router-dom';
import AppHeader from './components/AppHeader';
import Clipboard from './components/Clipboard';
import About from './components/About';
import { createInstance, MatomoProvider, useMatomo } from '@datapunt/matomo-tracker-react';
import { socket, SocketContext } from './context/SocketContext';
import Status from './components/Status';
import useSocketClient from './hooks/useSocketClient';

const muiThemeV0 = getMuiTheme({
  palette: {
    primary1Color: orange600,
    accent1Color: orange800,
  },
  tabs: {
    backgroundColor: blueGrey400,
  },
});

const muiTheme = createMuiTheme({
  palette: {
    primary: {
      main: orange600,
    },
    secondary: {
      main: orange800,
    }
  },
  tabs: {
    backgroundColor: blueGrey400,
  },
});

const style = {
  footer: {
    color: blueGrey400,
    margin: 10,
    paddingLeft: 10,
  },
  a: {
    color: blueGrey400,
  },
  background: {
    backgroundColor: blueGrey400,
    paddingBottom: 10,
    height: 240,
  }
};

const matomoInstance = createInstance({
  urlBase: 'https://matomo.trafex.nl',
  siteId: 2,
  linkTracking: false
});

ReactGA.initialize(process.env.REACT_APP_GA_CODE, {
  gaOptions: {
    siteSpeedSampleRate: 10
  }
});

const App = () => {

  const { trackPageView, enableLinkTracking } = useMatomo()
  enableLinkTracking();

  const history = useHistory();

  // Track page view
  useEffect(() => {
    trackPageView()
    history.listen(trackPageView) // To track the subsequent pageviews
  }, [history, trackPageView])

  useEffect(() => {
    const trackPage = () => {
      ReactGA.set({ page: history.location.pathname });
      ReactGA.pageview(history.location.pathname);
    };

    trackPage();
    history.listen(trackPage); // To track the subsequent pageviews
  }, [history]);

  const { ownRoomNumber, connectedToRoom, status, usersInRoom, messages } = useSocketClient(socket);

  return (
    <MuiThemeProvider theme={muiTheme}>
      <V0MuiThemeProvider muiTheme={muiThemeV0}>
        <MatomoProvider value={matomoInstance}>
          <SocketContext.Provider value={socket}>
            <div style={style.background}>
              <Helmet>
                <link rel="manifest" href="/manifest-v1.json" />
              </Helmet>
              <AppHeader />
              <Status
                status={status}
              />
              <Switch>
                <Route exact path="/">
                  <Clipboard ownRoomNumber={ownRoomNumber} connectedToRoom={connectedToRoom} usersInRoom={usersInRoom} messages={messages} />
                </Route>
                <Route path="/index.html">
                  <Clipboard ownRoomNumber={ownRoomNumber} connectedToRoom={connectedToRoom} usersInRoom={usersInRoom} messages={messages} />
                </Route>
                <Route path="/about.html">
                  <About />
                </Route>
              </Switch>
              <footer style={style.footer}>
                <a href="https://github.com/TrafeX/clipboard.ninja/blob/master/PRIVACY.md" style={style.a} target="_blank" rel="noopener noreferrer">Privacy Policy</a> - <a href="https://github.com/trafex/clipboard.ninja" style={style.a} target="_blank" rel="noopener noreferrer">Source on GitHub</a> - Created by <a href="https://www.trafex.nl" style={style.a} target="_blank" rel="noopener noreferrer">Tim de Pater</a>
              </footer>
            </div>
          </SocketContext.Provider>
        </MatomoProvider>
      </V0MuiThemeProvider>
    </MuiThemeProvider>
  )
};

export default App;
