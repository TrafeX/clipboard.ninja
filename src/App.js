// @flow
import React from 'react';
import {orange600, orange800, blueGrey400} from 'material-ui/styles/colors';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import {Helmet} from "react-helmet";
import {Route, Switch} from 'react-router-dom';
import AppHeader from './AppHeader';
import Clipboard from './Clipboard';
import About from './About';
import withTracker from './withTracker';

const muiTheme = getMuiTheme({
  palette: {
    primary1Color: orange600,
    accent1Color: orange800,
  },
  tabs: {
    backgroundColor: blueGrey400,
  },
});

const style = {
  footer: {
    color: blueGrey400,
    margin: 10,
    paddingLeft: 10,
  },
  a: {
    color: blueGrey400,
  },
  background: {
    backgroundColor: blueGrey400,
    paddingBottom: 10,
    height: 240,
  }
};

const App = () => (
  <MuiThemeProvider muiTheme={muiTheme}>
    <div style={style.background}>
      <Helmet>
        <link rel="manifest" href="/manifest-v1.json" />
      </Helmet>
      <AppHeader/>
      <Switch>
        <Route exact path="/" component={withTracker(Clipboard)}/>
        <Route path="/index.html" component={withTracker(Clipboard)}/>
        <Route path="/about.html" component={withTracker(About)}/>
      </Switch>
      <footer style={style.footer}>
        <a href="https://github.com/TrafeX/clipboard.ninja/blob/master/PRIVACY.md" style={style.a}>Privacy Policy</a> - Created by <a href="https://www.trafex.nl" style={style.a}>Tim de Pater</a> - Check out the <a href="https://github.com/trafex/clipboard.ninja" style={style.a}>source on GitHub</a>
      </footer>
    </div>
  </MuiThemeProvider>
);

export default App;
