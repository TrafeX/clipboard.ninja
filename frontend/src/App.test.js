import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

jest.mock('react-ga');

it('renders without crashing', () => {
  const wrapper = shallow(<App />).find('div#main').shallow();
  expect(wrapper).toMatchSnapshot();
});
