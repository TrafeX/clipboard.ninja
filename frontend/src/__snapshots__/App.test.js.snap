// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  id="main"
  style={
    Object {
      "backgroundColor": "#78909c",
      "height": 240,
      "paddingBottom": 10,
    }
  }
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <link
      href="/manifest-v1.json"
      rel="manifest"
    />
  </HelmetWrapper>
  <withRouter(AppHeader) />
  <Status
    status=""
  />
  <Switch>
    <Route
      exact={true}
      path="/"
    >
      <Clipboard
        connectedToRoom={null}
        messages={Array []}
        ownRoomNumber={null}
        usersInRoom={0}
      />
    </Route>
    <Route
      path="/index.html"
    >
      <Clipboard
        connectedToRoom={null}
        messages={Array []}
        ownRoomNumber={null}
        usersInRoom={0}
      />
    </Route>
    <Route
      path="/about.html"
    >
      <About />
    </Route>
  </Switch>
  <footer
    style={
      Object {
        "color": "#78909c",
        "margin": 10,
        "paddingLeft": 10,
      }
    }
  >
    <a
      href="https://github.com/TrafeX/clipboard.ninja/blob/master/PRIVACY.md"
      rel="noopener noreferrer"
      style={
        Object {
          "color": "#78909c",
        }
      }
      target="_blank"
    >
      Privacy Policy
    </a>
     - 
    <a
      href="https://github.com/trafex/clipboard.ninja"
      rel="noopener noreferrer"
      style={
        Object {
          "color": "#78909c",
        }
      }
      target="_blank"
    >
      Source on GitHub
    </a>
     - Created by 
    <a
      href="https://www.timdepater.com"
      rel="noopener noreferrer"
      style={
        Object {
          "color": "#78909c",
        }
      }
      target="_blank"
    >
      Tim de Pater
    </a>
  </footer>
</div>
`;
